File: tickets-service/internal/models/payment.go
package models

// Payment represents a payment entity with its attributes.
type Payment struct {
	UUID              string        `json:"uuid"`
	TicketID          string        `json:"ticket_id"`
	Amount            int64         `json:"amount"`
	PaymentMethod     PaymentMethod `json:"payment_method"`
	ProviderReference string        `json:"provider_reference"`
}

type PaymentMethod string

const (
	PaymentMethodCreditCard   PaymentMethod = "credit_card"
	PaymentMethodPayPal       PaymentMethod = "banknotes"
	PaymentMethodBankTransfer PaymentMethod = "bank_transfer"
)

File: frontend/src/modules/tickets/types/payment.ts
export interface Payment {
    uuid: string;
    ticket_id: string;
    amount: number;
    payment_method: "credit_card" | "banknotes" | "bank_transfer";
    provider_reference: string;
}

export interface CreatePaymentRequest {
    ticket_id: string;
    amount: number;
    payment_method: "credit_card" | "banknotes" | "bank_transfer";
    provider_reference: string;
}

export interface UpdatePaymentRequest extends CreatePaymentRequest {}
